samples/openvx/no_wrappers.cpp

```diff
@@ -0,0 +1,429 @@
+    vx_scalar shift = vxCreateScalar(context, VX_TYPE_INT32, &shiftVal);
+
+    // Sequence of meaningless image operations
+    vx_node nodes[] = {
+        vxGaussian3x3Node(graph, inputImage, gb),
+        vxSobel3x3Node(graph, gb, gx, gy),
+        vxMagnitudeNode(graph, gx, gy, mag),
+        vxPhaseNode(graph, gx, gy, ang),
+        vxConvertDepthNode(graph, mag, mag8, VX_CONVERT_POLICY_SATURATE, shift),
+        vxThresholdNode(graph, ang,  threshAng, thra),
+        vxThresholdNode(graph, mag8, threshMag, thrm),
+        vxAndNode(graph, thra, thrm, thrv),
+        vxAndNode(graph, inputImage, thrv, outputImage),
+    };
+
+    for (size_t i = 0; i < sizeof(nodes) / sizeof(vx_node); i++)
+    {
+        status = vxGetStatus((vx_reference)nodes[i]);
+        if (status != VX_SUCCESS) return status;
+    }
+
+    // Graph verification
+    status = vxVerifyGraph(graph);
+    return status;
+}
+
+
+int ovxDemo(std::string inputPath, bool useCopy)
+{
+    cv::Mat image = cv::imread(inputPath, cv::IMREAD_GRAYSCALE);
+    if (image.empty()) return -1;
+
+    //check image format
+    if (image.depth() != CV_8U || image.channels() != 1) return -1;
+
+    vx_status status;
+    vx_context context = vxCreateContext();
+    status = vxGetStatus((vx_reference)context);
+    if (status != VX_SUCCESS) return status;
+
+    //put user data from cv::Mat to vx_image
+    vx_image ovxImage;
+    ovxImage = convertCvMatToVxImage(context, image, useCopy);
+
+    vx_uint32 width = image.cols, height = image.rows;
+
+    vx_image ovxResult;
+    cv::Mat output;
+    if (useCopy)
+    {
+        //we will copy data from vx_image to cv::Mat
+        ovxResult = ovx::createImage(context, width, height, VX_DF_IMAGE_U8);
+    }
+    else
+    {
+        //create vx_image based on user data, no copying required
+        output = cv::Mat(height, width, CV_8U, cv::Scalar(0));
+        ovxResult = convertCvMatToVxImage(context, output, false);
+    }
+
+    vx_graph graph;
+    status = createProcessingGraph(context, ovxImage, ovxResult, graph);
+    if (status != VX_SUCCESS) return status;
+
+    // Graph execution
+    status = vxProcessGraph(graph);
+    if (status != VX_SUCCESS) return status;
+
+    //getting resulting image in cv::Mat
+    if (useCopy)
+    {
+        output = copyVxImageToCvMat(ovxResult);
+    }
+    else
+    {
+        //we should take user memory back from vx_image before using it (even before reading)
+        swapVxImage(ovxResult);
+    }
+
+    //here output goes
+    cv::imshow("processing result", output);
+    cv::waitKey(0);
+
+    //we need to take user memory back before releasing the image
+    if (!useCopy)
+        swapVxImage(ovxImage);
+
+    cv::destroyAllWindows();
+
+    status = vxReleaseContext(&context);
+    return status;
+}
+
+
+int main(int argc, char *argv[])
+{
+    const std::string keys =
+        "{help h usage ? | | }"
+        "{@image_name    | | image to be processed}"
+        "{nocopy         | | }"
```

Comment:
I suggest changing the option name to either 'copy' or 'zero_copy' and set a default value to it.